tests/test_fakes.py

```diff
@@ -1312,6 +1312,17 @@ def test_values(conn: snowflake.connector.SnowflakeConnection):
         ]
 
 
+def test_json_extract_cast_as_varchar(dcur: snowflake.connector.cursor.DictCursor):
+    dcur.execute("CREATE TABLE example (j VARIANT)")
+    dcur.execute("""INSERT INTO example VALUES ('{"str": "100", "number" : 100}')""")
```

Comment:
I get this error on a real Snowflake instance:
> ProgrammingError: 002023 (22000): SQL compilation error:
Expression type does not match column data type, expecting VARIANT but got VARCHAR(30) for column J